### PR TITLE
Single-agent env wrapper

### PR DESCRIPTION
**Pull request by Adaickalavan**
1. Converts a single-agent SMARTS environment's `step` and `reset` output to be compliant with gym spaces.
2. Helps users to easily train or evaluate SMARTS with existing single-agent reinforcement-learning algorithms.

### CODE REVIEW

File: smarts/env/wrappers/single_agent.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from typing import Any, Tuple
+
+import gym
+
+
+class SingleAgent(gym.Wrapper):
```
Comment (Gamenot) [inline smarts/env/wrappers/single_agent.py:28]: OK, I think we will want to use this explicitly in the [single_agent.py](https://github.com/huawei-noah/SMARTS/blob/wrapper-single-agent/examples/single_agent.py) example.

**Review by Gamenot — COMMENTED** [review]
Also this is related to #945.

File: smarts/env/wrappers/single_agent.py (hunk 1 above)
Comment (Adaickalavan) [reply smarts/env/wrappers/single_agent.py:28]: Currently, `single_agent.py` illustrates the multi-agent interface of SMARTS. The `episode.record_step()` on line 67 works on dictionary inputs for multi-agents. 

I attempted to change `single_agent.py` to use `SingleAgent` wrapper. Using `SingleAgent` wrapper makes the code look like normal single-agent gym-compliant code. But line 67 will then be written as:
episode.record_step({AGENT_ID:obs}, {AGENT_ID:reward}, {AGENT_ID:done, “__all__”:done}, {AGENT_ID:info}) 

1) So should we proceed with using `SingleAgent` wrapper in `single_agent.py` example? 
2) If we are using `SingleAgent` wrapper in `single_agent.py`, shall we just remove the `episode.record_step()` functionality in order to streamline the example?

Comment (Gamenot) [reply smarts/env/wrappers/single_agent.py:28]: > 1. Should we proceed with using SingleAgent wrapper in single_agent.py example?

Yes, I think we should still do so.

> 2. If we are using `SingleAgent` wrapper in `single_agent.py`, shall we just remove the` episode.record_step()` functionality in order to streamline the example?

The use of record step is just for visual feedback when running the example. I am not sure if we can get rid of it, I could suggest we can make an overload or simplified forwarding version of `record_step()`.

Comment (Adaickalavan) [reply smarts/env/wrappers/single_agent.py:28]: Introduced `SingleAgent` wrapper into `single_agent.py` example.

**Review by Gamenot — APPROVED** [review]

**Review by sah-huawei — APPROVED** [review]

**Pull request merged**